app/controller/LayerTreeController.js

```diff
@@ -293,19 +291,24 @@ Ext.define('CpsiMapview.controller.LayerTreeController', {
         var me = this;
 
         if (pressed) {
-            this.addWmsWindow = Ext.create('CpsiMapview.view.window.MinimizableWindow', {
-                items: [
-                    {
-                        xtype: 'cmv_add_wms_form'
-                    }
-                ],
-            });
+            if (!this.addWmsWindow) {
+                this.addWmsWindow = Ext.create(me.getView().addWmsWindowConfig);
+                this.addWmsWindow.on('hide', function () {
+                    button.setPressed(false);
+                });
+            }
             this.addWmsWindow.show();
-            this.addWmsWindow.on('close', function () {
-                button.setPressed(false);
-                me.addWmsWindow = null;
-            });
+
         } else if (this.addWmsWindow) {
+            this.addWmsWindow.hide();
```

Comment:
For this to work the `closeAction` needs always to be `'hide'`. Maybe it should be fixed to that?

Reply:
Good point - changed back to `close` and will let the closeAction handle that.